tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -4463,5 +4464,44 @@ std::optional<Value> convertSinOp(PatternRewriter& rewriter, Operation* op,
       .getResult();
 }
 
+// Lowers Sign operator to a sequence of TOSA ops.
+llvm::Optional<Value> convertSignOp(PatternRewriter& rewriter, Operation* op,
+                                    Value input, RankedTensorType output_type) {
+  if (output_type.getElementType().isa<mlir::quant::QuantizedType>()) {
+    (void)rewriter.notifyMatchFailure(op, "tfl quantization not yet supported");
+    return llvm::None;
+  }
+
+  // TOSA greater and select can both broadcast, so simply create a tensor with
+  // one element.
+  Value pos_one, neg_one, zero;
+  if (output_type.getElementType().isa<FloatType>()) {
+    pos_one = getTosaConstTensorSingleF32(rewriter, op, 1.0f);
+    neg_one = getTosaConstTensorSingleF32(rewriter, op, -1.0f);
+    zero = getTosaConstTensorSingleF32(rewriter, op, 0.0f);
+  } else {
+    pos_one = getTosaConstTensorSingleI32(rewriter, op, 1);
+    neg_one = getTosaConstTensorSingleI32(rewriter, op, -1);
```

Comment:
This `getTosaConstTensorSingleI32` operators should be `getTosaConstTensorScalarInt`, and instead take a `Type` for the element type to define the expected bitwidth. This way one implementation can handle all integer bitwidths.

Reply:
I like it this way. Combine this with your comment about using ImplicitLocOpBuilder as argument, `op` can also be removed from the list (it is just used for getting the location).
If you think this new function is ok, we can do a follow-up patch to substitute `getTosaConstTensorScalarInt` for `getTosaConstTensorSingleI32` for the rest of tosa code.